### ISSUE TITLE
Please replace deprecated CloseNotify with Context().Done()

### ISSUE THREAD

**Issue by kondratev** [jcuga/golongpoll#21]
disconnectNotify := r.Context().Done()
```diff --git a/longpoll.go b/longpoll.go
--- a/longpoll.go
+++ b/longpoll.go
@@ -270,7 +270,7 @@ func getLongPollSubscriptionHandler(maxTimeoutSeconds int, subscriptionRequests
                // event that a client crashes or the connection goes down.  We don't
                // need to wait around to fulfill a subscription if no one is going to
                // receive it
-               disconnectNotify := w.(http.CloseNotifier).CloseNotify()
+               disconnectNotify := r.Context().Done()
                select {
                case <-time.After(time.Duration(timeout) * time.Second):
                        // Lets the subscription manager know it can discard this request's
```

**Comment by jcuga** [comment]
Fixed by f48ad205860a571ff4b387178b62be2b2224f57f.

Note: your suggested change broke some unit tests that were using the old CloseNotify to simulate client disconnects.  I updated those tests to use `context.WithTimeout()` instead.

Also, I plan on releasing a new version of golongpoll in the next 2-4 weeks that:
* Fixes issue #19 
* Adds an official JavaScript client
* Updates the official go client
* Supports persisting events to file for reuse across program runs
* Supports custom AddOns that can add additional behavior to the LongpollManager.  This is how file persistence is added.
* Additional cleanup
